statistics/handle/update.go

```diff
@@ -332,9 +333,9 @@ func (h *Handle) dumpTableStatCountToKV(is infoschema.InfoSchema, physicalTableI
 			return
 		}
 		affectedRows += sctx.GetSessionVars().StmtCtx.AffectedRows()
-		// If only the partition is locked, we don't need to update the global-stats.
+		// If the partition is locked, we don't need to update the global-stats.
 		// We will update its global-stats when the partition is unlocked.
-		if isTableLocked || !isPartitionLocked {
+		if !isPartitionLocked {
```

Comment:
Looks like it should be `!isTableLocked && !isPartitionLocked`?

Reply:
I think there are only two cases for the global stats now:
1. The table id and the partition id are both not locked: just directly dump the delta.
2. Otherwise, we neither dump nor "stash" the delta for the table id, and we just "stash" the delta for the partition id. And when unlocking, delta will be dumped for both partition id and table id.

Reply:
I don't think we need this condition.

```go
		// If the partition is locked, we don't need to update the global-stats.
		// We will update its global-stats when the partition is unlocked.
		// 1. If table is locked and partition is locked, we only stash the delta in the partition's lock info.
		//    we will update its global-stats when the partition is unlocked.
		// 2. If table is locked and partition is not locked(new partition after lock), we only stash the delta in the table's lock info.
		//    we will update its global-stats when the table is unlocked. We don't need to specially handle this case.
		// 3. If table is not locked and partition is locked, we only stash the delta in the partition's lock info.
		//    we will update its global-stats when the partition is unlocked.
		// 4. If table is not locked and partition is not locked, we update the global-stats.
		// To sum up, we only need to update the global-stats when the partition is unlocked.
```

Reply:
Let me test it again.

Reply:
Updated! Thanks for your review! I am so buggy 😢